src/backend.rs

```diff
@@ -757,6 +808,23 @@ impl SharedBackend {
         }
     }
 
+    /// Returns any arbitrary request on the provider
+    pub fn do_any_request<T, F>(&mut self, fut: F) -> DatabaseResult<T>
+    where
+        F: Future<Output = Result<T, eyre::Report>> + Send + 'static,
+        T: fmt::Debug + Send + 'static,
+    {
+        self.blocking_mode.run(|| {
+            let (sender, rx) = oneshot_channel::<Result<T, eyre::Report>>();
+            let req = BackendRequest::AnyRequest(Box::new(AnyRequestFuture {
+                sender,
+                future: Box::pin(fut),
+            }));
+            self.backend.unbounded_send(req)?;
+            rx.recv()?.map_err(|err| DatabaseError::AnyRequest(Arc::new(err)))
+        })
+    }
```

Comment:
the way this is implemented kinda acts as `tokio::Handle::block_on` but it sends the request to the async backend, I think this could be quite useful, especially if we extend this with caching functionality, because rn this does not persist cache the responses.

we can add this feature separately though

Reply:
Will follow-up accordingly.